returnn/tf/network.py

```diff
@@ -2036,6 +2036,20 @@ def register_search_choices_for_beam(self, beam, search_choices):
       return root_net.register_search_choices_for_beam(beam, search_choices)
     self._map_search_beam_to_search_choices[beam] = search_choices
 
+  def iter_layers_recursively(self):
+    """
+    Yield all sub-layers of this network (including layers inside rec-layers)
+    """
+    from .layers.rec import RecLayer, _SubnetworkRecCell
+    for layer in self.layers.values():
+      yield layer
+      if isinstance(layer, RecLayer) and isinstance(layer.cell, _SubnetworkRecCell):
+        # TODO: replace with yield from once python2.7 support is dropped
+        for sublayer in layer.cell.net.iter_layers_recursively():
+          yield sublayer
+        for sublayer in layer.cell.input_layers_net.iter_layers_recursively():
+          yield sublayer
+
```

Comment:
You should also iterate over the output layers:
```suggestion
        for sublayer in layer.cell.output_layers_net.iter_layers_recursively():
          yield sublayer
          
```
or however you resolve albertz comment.